lib/rollbar/notifier.rb

```diff
@@ -496,6 +500,40 @@ def do_post(uri, body, access_token)
       handle_net_retries { http.request(request) }
     end
 
+    def http_proxy_for_em(uri)
+      proxy = http_proxy(uri)
+      {
+        :proxy => {
+          :host => proxy.host,
+          :port => proxy.port,
+          :authorization => [proxy.user, proxy.password]
+        }
+      }
+    end
+
+    def http_proxy(uri)
+      @http_proxy ||= proxy_from_config || proxy_from_env(uri) || null_proxy
+    end
+
+    def proxy_from_config
+      proxy = configuration.proxy
+      return nil unless proxy
+
+      px = URI.parse(proxy[:host])
+      px.port = proxy[:port]
+      px.user = proxy[:user]
+      px.password = proxy[:password]
+      px
+    end
+
+    def proxy_from_env(uri)
+      uri.find_proxy
+    end
+
+    def null_proxy
+      OpenStruct.new(:host => nil, :port => nil, :user => nil, :password => nil)
```

Comment:
To be honest, this started life as a Struct, then an abandoned refactoring changed it to an OpenStruct and I neglected to change it back.  Happy to do that.  It makes this code more concise.